### PR TITLE
Fix 163: Coin collector settings

### LINKED ISSUE #163: Issue with generating `coin collector` challenges
I am able to generate coin collector challenge maps using levels 0 to 99 (using `tw-make tw-coin_collector --level x` however when I try and use level 100 or higher there are two different error messages that appear. These will not appear all of the time (due to the random seed), but will appear quite frequently.

The first (which can be replicated with `tw-make tw-coin_collector --level 100 --seed 1`) is:

```
Traceback (most recent call last):
  File "/home/ben/miniconda3/envs/pytorch10/bin/tw-make", line 191, in <module>
    game = make_game(settings=args.__dict__, options=options)
  File "/home/ben/miniconda3/envs/pytorch10/lib/python3.7/site-packages/textworld/challenges/coin_collector.py", line 75, in make
    return make_game(distractor_mode, options)
  File "/home/ben/miniconda3/envs/pytorch10/lib/python3.7/site-packages/textworld/challenges/coin_collector.py", line 98, in make_game
    if mode == "simple" and float(options.nb_rooms) / options.quest_length > 4:
ZeroDivisionError: float division by zero
```

The second (which seems to only appear when level > 200, can be replicated with `tw-make tw-coin_collector --level 250 --seed 1`) is:

```
Traceback (most recent call last):
  File "/home/ben/miniconda3/envs/pytorch10/bin/tw-make", line 191, in <module>
    game = make_game(settings=args.__dict__, options=options)
  File "/home/ben/miniconda3/envs/pytorch10/lib/python3.7/site-packages/textworld/challenges/coin_collector.py", line 75, in make
    return make_game(distractor_mode, options)
  File "/home/ben/miniconda3/envs/pytorch10/lib/python3.7/site-packages/textworld/challenges/coin_collector.py", line 160, in make_game
    M.set_quest_from_commands(walkthrough)
  File "/home/ben/miniconda3/envs/pytorch10/lib/python3.7/site-packages/textworld/generator/maker.py", line 569, in set_quest_from_commands
    game_file = self.compile(pjoin(tmpdir, "record_quest.ulx"))
  File "/home/ben/miniconda3/envs/pytorch10/lib/python3.7/site-packages/textworld/generator/maker.py", line 713, in compile
    self._working_game = self.build()
  File "/home/ben/miniconda3/envs/pytorch10/lib/python3.7/site-packages/textworld/generator/maker.py", line 693, in build
    game.change_grammar(self.grammar)
  File "/home/ben/miniconda3/envs/pytorch10/lib/python3.7/site-packages/textworld/generator/game.py", line 398, in change_grammar
    generate_text_from_grammar(self, self.grammar)
  File "/home/ben/miniconda3/envs/pytorch10/lib/python3.7/site-packages/textworld/generator/text_generation.py", line 112, in generate_text_from_grammar
    assign_name_to_object(room, grammar, game.infos)
  File "/home/ben/miniconda3/envs/pytorch10/lib/python3.7/site-packages/textworld/generator/text_generation.py", line 82, in assign_name_to_object
    values = grammar.generate_name(obj.type, room_type=obj_infos.room_type, exclude=exclude)
  File "/home/ben/miniconda3/envs/pytorch10/lib/python3.7/site-packages/textworld/generator/text_grammar.py", line 348, in generate_name
    name, adj, noun = self.generate_name(obj_type, room_type, include_adj=True, exclude=exclude)
  File "/home/ben/miniconda3/envs/pytorch10/lib/python3.7/site-packages/textworld/generator/text_grammar.py", line 365, in generate_name
    raise ValueError(msg)
ValueError: Not enough variation for '#work_(r)#'. You can add more variation in the 'house' related grammar files located in '/home/ben/miniconda3/envs/pytorch10/lib/python3.7/site-packages/textworld/generator/data/text_grammars' or turn on the 'include_adj=True' grammar flag. In last resort, you could always turn on the 'allowed_variables_numbering=True' grammar flag to append unique number to object name.
```

### PR DESCRIPTION
**Pull request by MarcCote**
Currently, `tw-make tw-coin_collector --level 100` is trying to generate a game with quest length of 0 (which is incorrect). This PR fixes #163.

### CODE REVIEW

File: textworld/challenges/tests/test_coin_collector.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+
+def test_making_coin_collector():
+    for level in [1, 100, 101]:
+        options = textworld.GameOptions()
+        options.seeds = 1234
+
+        settings = {"level": level}
+        game = make(settings, options)
+        assert len(game.quests[0].commands) == (level - 1) % 100 + 1
```
[the remark this answers is not in the document]
Comment (MarcCote) [reply textworld/challenges/tests/test_coin_collector.py]: Totally make sense.

File: textworld/challenges/tests/test_coin_collector.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    # Not enough variation for 200 rooms.
+    options = textworld.GameOptions()
+    options.seeds = 1234
+    settings = {"level": 200}
+    assert_raises(ValueError, make, settings, options)
```
[the remark this answers is not in the document]
Comment (MarcCote) [reply textworld/challenges/tests/test_coin_collector.py]: Right, this also highly depends on the size of the default text grammar we have in the knowledge base. I don't know how to properly handle that error. Ideally, I would want it to be clear to the user why that's happening.

File: textworld/challenges/tests/test_coin_collector.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MarcCote) [reply textworld/challenges/tests/test_coin_collector.py]: Fixed

**Comment by tavianator** [comment]
The test failure is:
```
ERROR: test_coin_collector.test_making_coin_collector
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/travis/build/microsoft/TextWorld/textworld/challenges/tests/test_coin_collector.py", line 23, in test_making_coin_collector
    game = make(settings, options)
  File "/home/travis/build/microsoft/TextWorld/textworld/challenges/coin_collector.py", line 72, in make
    options.grammar.allowed_variables_numbering = settings["force_entity_numbering"]
KeyError: 'force_entity_numbering'
```

**Comment by MarcCote** [comment]
Not sure if it is the right way for enforcing the `--entity-numbering` but that's clear to the user.

**Comment by MarcCote** [comment]
Fixed.

**Pull request merged**